docker-compose-e2e-tests.yml

```diff
@@ -0,0 +1,67 @@
+version: "3.7"
+
+name: mit-open-e2e-tests
+
+services:
+  db:
+    image: postgres:12.17
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_PASSWORD=postgres
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
+      interval: 5s
+      timeout: 5s
+      retries: 2
+
+  nginx:
+    image: nginx:1.25.3
+    ports:
+      - 8063:8063
+    links:
+      - web
+    volumes:
+      - ./config/nginx.conf:/etc/nginx/conf.d/web.conf
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:8063"]
+      interval: 30s
+      timeout: 10s
+      retries: 5
+
+  web:
+    build:
+      context: .
+    environment:
+      DATABASE_URL: postgres://postgres:postgres@db:5432/e2e_postgres # pragma: allowlist secret
+      PORT: 8061
+    env_file: .env.ci
+    depends_on:
+      db:
+        condition: service_healthy
+    command: > # pragma: allowlist secret
+      bash -c "export PGPASSWORD=postgres
+      && dropdb --if-exists -h db -U postgres e2e_postgres
+      && createdb -h db -U postgres e2e_postgres
+      && python3 manage.py collectstatic --noinput --clear
+      && python3 manage.py migrate --noinput
+      && python3 manage.py loaddata e2e_testing/fixtures/*.json
+      && uwsgi uwsgi.ini --honour-stdin"
+    stdin_open: true
+    tty: true
+    ports:
+      - 8061:8061
+    links:
+      - db
```

Comment:
I'm not sure we want this, though I'm thinking in terms that it is desirable that the container doesn't change unless intentionally built.

An issue is that the front end needs to be built locally in production mode for the static assets to correctly load from a relative path when loaded from inside another container.

> Still getting timeout errors on the 3 tests

It could be that the issue you are seeing locally with the tests failing is that you have a local front end build pointing to `http://{MITOPEN_HOSTNAME}:{WEBPACK_PORT_MITOPEN}/`, so the HTML will try to load the JS bundle from e.g. od.odl.local, which won't exist on the Docker network.

On CI and locally, I need to first build the front end with `NODE_ENV=production`. Of course the `watch` service from the main `docker-compose.yml` mounts the root directory so it will overwrite the front end build with the development version when it runs. I've added a [note](https://github.com/mitodl/mit-open/blob/47d8e8875fcd0f3e45f7fed5c6d31431b4ddb7b8/e2e_testing/README.md?plain=1#L60) for this to the README.

This is only an issue if we want to run the e2e tests from inside their container, which we don't have to. We can start the main Docker Compose services normally and run the tests from local (though the web service would need to be using the `e2e_postgres` database):

```bash
yarn --cwd e2e_testing test
```

It would be convenient for the fixture files to be available to the container when changed on local, though I've instead provided a [script](https://github.com/mitodl/mit-open/blob/jk/418-e2e-tests-ci/e2e_testing/scripts/apply-fixtures.sh) that copies them over and applies them.